### PR TITLE
Move LinalgQuantized* passes to GlobalOptimization

### PR DESCRIPTION
**Pull request by qedawkins**
These passes are a better fit for GlobalOptimization given they have to
do with Linalg ops. Additionally this turns on the passes for all
frontends now that other frontends are starting to generate these ops
too.

### CODE REVIEW

**Review by hanhanW — COMMENTED** [review]
Can we remove the passes from `TOSA/InputConversion/Passes.cpp`? I think adding it the common input conversion pipeline should cover it right?

https://github.com/iree-org/iree/blob/95d556227fe2df27a0337de2507b189ef855bbc6/compiler/plugins/input/TOSA/InputConversion/Passes.cpp#L68-L72

**Comment by qedawkins** [comment]
Removed the passes from TOSA, and also moved the passes to GlobalOptimization.

File: compiler/src/iree/compiler/GlobalOptimization/Utils.cpp [hunk 1]
```diff
@@ -121,5 +121,67 @@ wrapConsecutiveOpsInDispatchRegion(RewriterBase &rewriter,
   return IREE::Flow::movePrecedingOpsIntoDispatchRegion(rewriter, precedingOps,
                                                         regionOp);
 }
+// Reduce the input value along the reduction dimensions
+Value sumReduceDimensionSubset(ImplicitLocOpBuilder &rewriter, Value val,
+                               Type accETy, ArrayRef<bool> is_reduction) {
```
Comment (hanhanW) [inline compiler/src/iree/compiler/GlobalOptimization/Utils.cpp]: nit: Move the function comment to where it is declared (i.e., Passes.h). Also, lets have a blank line between functions. It usually improves readability.

```suggestion

Value sumReduceDimensionSubset(ImplicitLocOpBuilder &rewriter, Value val,
                               Type accETy, ArrayRef<bool> is_reduction) {
```

**Review by hanhanW — APPROVED** [review]
nice, just one nit!

File: compiler/src/iree/compiler/GlobalOptimization/Passes.cpp [hunk 2]
```diff
@@ -91,6 +91,9 @@ void buildGlobalOptimizationPassPipeline(
 
   // Preprocessing passes to get the program into a canonical state.
   FunctionLikeNest(mainPassManager)
+      .addPass(createLinalgQuantizedConvToConvPass)
+      .addPass(createLinalgQuantizedMatmulToMatmulPass)
+      .addPass(IREE::Flow::createCanonicalizerPass)
```
Comment (MaheshRavishankar) [inline compiler/src/iree/compiler/GlobalOptimization/Passes.cpp:96]: Not a great idea to have flow pass used every where. Maybe move this to `Util` dialect or something.

**Review by MaheshRavishankar — APPROVED** [review]

File: compiler/src/iree/compiler/GlobalOptimization/Passes.cpp (hunk 2 above)
Comment (qedawkins) [reply compiler/src/iree/compiler/GlobalOptimization/Passes.cpp:96]: I think Ian is moving the dispatch region formation pipeline somewhere else and I think this pass will go with it. Let's see where it lands after that move.

**Pull request merged**